File: evidence/generated-systems/golang/governance-on-golang/governance-on-golang/internal/model/RoleAssignment.go
package model

import (
    "time"
    "gorm.io/gorm"
)

//==============================================================
// RoleAssignment Declaration
//==============================================================
type RoleAssignment struct {
    gorm.Model
     EffectiveFrom                                                            time.Time
    EffectiveTo                                                            time.Time
    PersonId         *uint
    Person           *Person `gorm:"foreignKey:PersonId"`
    RoleId         *uint
    Role           *Role `gorm:"foreignKey:RoleId"`
    GovernanceBodyId         *uint
    GovernanceBody           *GovernanceBody `gorm:"foreignKey:GovernanceBodyId"`
    OrganizationId         *uint
    Organization           *Organization `gorm:"foreignKey:OrganizationId"`

// parent associations as their child

}

File: evidence/generated-systems/django/governance-on-django/governanceOnDjango/models/RoleAssignment.py
from django.db import models

#======================================================================
# 
# Encapsulates data for model RoleAssignment
#
# @author Harbormaster Dev Team
#
#======================================================================

#======================================================================
# Class RoleAssignment Declaration
#======================================================================
class RoleAssignment (models.Model):

#======================================================================
# attribute declarations
#======================================================================
	effectiveFrom = models.DateField(null=True)
	effectiveTo = models.DateField(null=True)
	person = models.ForeignKey('Person', on_delete=models.CASCADE, null=True, blank=True, related_name='+')
	role = models.ForeignKey('Role', on_delete=models.CASCADE, null=True, blank=True, related_name='+')
	governanceBody = models.ForeignKey('GovernanceBody', on_delete=models.CASCADE, null=True, blank=True, related_name='+')
	organization = models.ForeignKey('Organization', on_delete=models.CASCADE, null=True, blank=True, related_name='+')

#======================================================================
# function declarations
#======================================================================
	def toString(self):
		str = ""
		str = str + self.effectiveFrom
		str = str + self.effectiveTo
		return str;
    
	def __str__(self):
		return self.toString();

	def identity(self):
		return "RoleAssignment";
    
	def objectType(self):
		return "RoleAssignment";
